### PR TITLE
Include information about CVE patches in CycloneDX output

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

**Comment by henrirosten** [comment]
@raboof : thanks for the PR. 
Let me know if/when you want a review on this change.

**Comment by raboof** [comment]
> @raboof : thanks for the PR. Let me know if/when you want a review on this change.

I think it's ready for a review now.

I didn't add any explicit test, but I did check that the existing test that generates an SBOM for `hello` also contains dependencies that have security patches, and that test still succeeds (indicating the generated sections conform to the schema).

File: src/sbomnix/sbomdb.py [hunk 1]
```diff
@@ -383,6 +383,21 @@ def _drv_to_cdx_component(drv, uid="store_path"):
     if "meta_description" in drv._asdict() and drv.meta_description:
         component["description"] = drv.meta_description
     _cdx_component_add_licenses(component, drv)
+    if drv.patches:
+        security_patches = []
+        for p in drv.patches.split(" "):
+            print(p)
+            m = re.search(r"CVE-\d{4}-\d+", p)
```
Comment (henrirosten) [inline src/sbomnix/sbomdb.py:390]: Unless there's a good reason not to, I would ignre the case in this match in case the data is dirty.

**Review by henrirosten — CHANGES_REQUESTED** [review]
Thanks for this PR, see the below comments.
Also, can you add a short description to this PR explaining why this change is needed?

File: src/sbomnix/sbomdb.py [hunk 2]
```diff
@@ -383,6 +383,21 @@ def _drv_to_cdx_component(drv, uid="store_path"):
     if "meta_description" in drv._asdict() and drv.meta_description:
         component["description"] = drv.meta_description
     _cdx_component_add_licenses(component, drv)
+    if drv.patches:
+        security_patches = []
+        for p in drv.patches.split(" "):
+            print(p)
```
[the remark this answers is not in the document]
Comment (raboof) [reply src/sbomnix/sbomdb.py]: oops :heavy_check_mark:

**Review by henrirosten — APPROVED** [review]

**Comment by henrirosten** [comment]
Thanks, merged.

**Pull request merged**